### ISSUE TITLE
Command line argument parsing breaks on non-.net platforms

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn#9954]
**Version Used**:  1.2.0.60220

**Steps to Reproduce**:

Using Mono to run csc.exe on Windows uses following syntax

1.Run `C:\Program Files (x86)\Mono\bin>mono-sgen.exe --attach=disable ..\..\..\roslyn\tools\csc.exe`

**Expected Behavior**:

Correctly parsed arguments of csc and not all arguments passed to mono-sgen.exe. This is issue in Roslyn codebase which assumes `csc.exe` is the process which in Mono case it's not true.

Relevant source code https://github.com/dotnet/roslyn/blob/df01b581f1e224d0e34418724679afeae3c5fa76/src/Compilers/Core/CommandLine/BuildClient.cs#L221

This is similar to #6677 except for Mono on Windows this is a blocker.

**Actual Behavior**:

Microsoft (R) Visual C# Compiler version 1.2.0.60220
Copyright (C) Microsoft Corporation. All rights reserved.

error CS2007: Unrecognized option: '--attach=disable'

**Comment by jaredpar** [comment]
Believe this is the issue we discussed in person. 

The root problem here is that csc essentially ignores the arguments passed to `Main`.  Instead it calls into Win32 to get the original command line argument (single string) and hand parses it.  This is the behavior of the native compiler and had to be emulated for back compat reasons. 

The conclusion we came to is that we only have to do this for back compat scenarios.  Hence we should limit this to when csc is run under the desktop CLR on Windows.  In all other cases just accept the command line as given and process it.
